routes/authorize.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE.txt in the project root for license information.
+var express = require('express');
+var router = express.Router();
+var authHelper = require('../helpers/auth');
+
+// Get authroize route
+// Handles returned authentication code from Microsoft's OAuth2 login
+router.get('/', async function(req, res, next) {
+  // Get auth code
+  const code = req.query.code;
+
+  // If code is present, use it
+  if (code) {
+    let token;
+
+    try {
+      token = await authHelper.getTokenFromCode(code, res);
+    } catch (error) {
+      res.json({ title: 'Error', message: 'Error exchanging code for token', error: error });
+    }
+
+    // Redirect to home
+    res.redirect('/');
```

Comment:
I would move this into the try block. Moving in there allows you to remove the `else` section as it is implied. Potentially turning this into error first approach would help to. Check if there is no code, then deal with that error case. Then you can deal with success.